### PR TITLE
Add post-genesis deposit testing to long-running E2E

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR adds deposit testing support to E2E to fully test validator deposits and activation.

Due to the time it takes to test this, it is only run as part of the "Long Running" E2E, now split up into its own file.

If E2E_EPOCHS is not set, the test will not run. Testing show allow for at least 16 epochs to ensure deposited validators are performing correctly.

### CODE REVIEW

File: endtoend/evaluators/validator.go [hunk 1]
```diff
@@ -46,6 +49,13 @@ func afterNthEpoch(afterEpoch uint64) func(uint64) bool {
 	}
 }
 
+// Not including first epoch because of issues with genesis.
+func betweenEpochs(fromEpoch uint64, toEpoch uint64) func(uint64) bool {
```
Comment (prestonvanloon) [inline endtoend/evaluators/validator.go:53]: please change to boolean type name

```suggestion
func isBetweenEpochs(fromEpoch uint64, toEpoch uint64) func(uint64) bool {
```

File: endtoend/long_minimal_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+func TestEndToEnd_Long_MinimalConfig(t *testing.T) {
+	testutil.ResetCache()
+	params.UseMinimalConfig()
+
+	epochsToRun := 20
+	var err error
+	if epochs, ok := os.LookupEnv("E2E_EPOCHS"); ok {
+		if !ok {
+			return
```
Comment (prestonvanloon) [inline endtoend/long_minimal_e2e_test.go:23]:
```suggestion
			t.Skip("E2E_EPOCHS not set")
```

File: endtoend/long_minimal_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+func TestEndToEnd_Long_MinimalConfig(t *testing.T) {
+	testutil.ResetCache()
+	params.UseMinimalConfig()
+
+	epochsToRun := 20
+	var err error
+	if epochs, ok := os.LookupEnv("E2E_EPOCHS"); ok {
+		if !ok {
+			t.Skip("E2E EPOCHS not set")
```
Comment (prestonvanloon) [inline endtoend/long_minimal_e2e_test.go:23]:
```suggestion
			t.Skip("E2E_EPOCHS not set")
```

**Comment by rauljordan** [comment]
@0xKiwi can we open future PR's into prysm directly so that bulldozer can auto-update? (it doesn't always work, but it can help sometimes). @prestonvanloon ready for another look

**Pull request merged**